### ISSUE TITLE
Re-implement repo with full git commit history

### ISSUE THREAD

**Issue by mitchellcash** [ionomy/ion#21]
Before we get too far along, I believe we should re-implement this project with a the full git commit history from the first commit in Bitcoin, to the Dash/Blackcoin fork (I'm not sure the actual timeline of how we got to ion).

Would this be possible for the lead coin dev to do something like this?

1. Create a new branch
2. Remove all history and code
3. From whatever point in whatever coin repository he started from to import that full git history and code
4. Cherry-pick all the commits to Ion Core to date (at time of writing 30)
5. Merge this over master replacing the current 30 commit history with the full commit history from the start of Bitcoin to Ion

**Comment by IngCr3at1on** [comment]
Get out of my head! Though simply rebasing might be easier (depending).

**Comment by mitchellcash** [comment]
@IngCr3at1on I figured there are probably better ways to do this :P 

But I think this needs to be done as you stated:
- Totally goes against what is generally considered proper open source practices
- There's a reason revision control systems exist and nuking the history isn't one of them

Even if the lead coin dev can let us know the exact commit ID and repo we forked from I can give this a shot.

**Comment by RdeWilde** [comment]
I agree. This way we can more easily keep in sync with development and fixes on the core.

**Comment by adammatlack** [comment]
yup this shouldn't be a problem, im not sure why it was left out honestly

**Comment by IonomyGuy** [comment]
Honestly i was givin a zip file with the source. That is why i made a clean repo.

**Comment by IonomyGuy** [comment]
If anyone want to tell me exactly how to do this i will rebase this.

**Comment by IngCr3at1on** [comment]
Why does that not surprise me lol... @IonomyGuy there's a chance there will be a couple of conflicts during a rebase so it may not be as smooth of a process as one might hope... However the general idea is as follows.

`git remote add upstream <upstream coin repo address>`
`git upstream fetch --no-tags`
`git checkout upstream/master`
`git checkout -b oldmaster`
`git checkout master`
`git rebase oldmaster`

That would rebase the current master branch over the oldmaster branch...
For sanity sake it may be safer to not rebase the master branch lol though so this may be better...

after `git checkout master` before `git rebase oldmaster`
`git checkout -b newmaster`
then do the `git rebase oldmaster` after that.

Basically that would checkout a new branch based on the current master branch which you would then rebase over the upstream master branch. After that you could push that as a separate branch to review, etc before actually replacing the current master branch with the newly rebased master branch.

**Comment by IonomyGuy** [comment]
Well then i need to figure out where the master devs pulled from. Then i can do this.

**Comment by mitchellcash** [comment]
@IonomyGuy nice work! Look forward to seeing this, let us know if we can be of any help.
